mlops/custom_objectdetection/src/mlops_pipeline.py

```diff
@@ -0,0 +1,348 @@
+    }
+
+
+def construct_pipeline(
+    cluster_name: str,
+    environment_name: str,
+    display_name: str,
+    build_environment: str,
+    build_reference: str,
+    model_name: str,
+    dataset_name: str,
+    ml_client,
+):
+    """
+    Construct a pipeline job for NYC taxi data regression.
+
+    Args:
+        cluster_name (str): The name of the cluster to use for pipeline execution.
+        environment_name (str): The name of the environment to use for pipeline execution.
+        display_name (str): The display name of the pipeline job.
+        build_environment (str): The environment to deploy the pipeline job.
+        build_reference (str): The build reference for the pipeline job.
+        model_name (str): The name of the model.
+        dataset_name (str): The name of the dataset.
+        ml_client: The machine learning client.
+
+    Returns:
+        pipeline_job: The constructed pipeline job.
+    """
+    registered_data_asset = ml_client.data.get(name=dataset_name, label="latest")
+
+    parent_dir = os.path.join(os.getcwd(), "mlops/custom_objectdetection/components")
+
+    prepare_data = load_component(source=parent_dir + "/prep.yml")
+    transform_data = load_component(source=parent_dir + "/transform.yml")
+    train_model = load_component(source=parent_dir + "/train.yml")
+    predict_result = load_component(source=parent_dir + "/predict.yml")
+    score_data = load_component(source=parent_dir + "/score.yml")
+    register_model = load_component(source=parent_dir + "/register.yml")
+
+    # Set the environment name to custom environment using name and version number
+    prepare_data.environment = environment_name
+    transform_data.environment = environment_name
+    train_model.environment = environment_name
+    predict_result.environment = environment_name
+    score_data.environment = environment_name
+    register_model.environment = environment_name
+
+    gl_pipeline_components.append(prepare_data)
+    gl_pipeline_components.append(transform_data)
+    gl_pipeline_components.append(train_model)
+    gl_pipeline_components.append(predict_result)
+    gl_pipeline_components.append(score_data)
+    gl_pipeline_components.append(register_model)
+
+    pipeline_job = custom_objectdetection_data_regression(
+        Input(type="uri_folder", path=registered_data_asset.id),
+        model_name,
+        build_reference,
+    )
+
+    pipeline_job.display_name = display_name
+    pipeline_job.tags = {
+        "environment": build_environment,
+        "build_reference": build_reference,
+    }
+
+    # demo how to change pipeline output settings
+    pipeline_job.outputs.pipeline_job_prepped_data.mode = "rw_mount"
+
+    # set pipeline level compute
+    pipeline_job.settings.default_compute = cluster_name
+    pipeline_job.settings.force_rerun = True
+    # set pipeline level datastore
+    pipeline_job.settings.default_datastore = "workspaceblobstore"
+
+    return pipeline_job
+
+
+def execute_pipeline(
```

Comment:
I know we have a story in the backlog, so just treat this comment as me expressing my excitement once it gets moved into the `mlops/common` directory 👍